src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -213,15 +205,22 @@ struct OperationViewModel {
     }
 }
 
-private func filterParams(for params: [Parameter]?, with allowed: [ParameterLocation]) -> [Parameter] {
-    let optionsParams = params?.filter { param in
-        if let httpParam = param.protocol.http as? HttpParameter {
-            return allowed.contains(httpParam.in)
+private func sort(params: [ParameterType]?) -> (method: [ParameterType], optional: [ParameterType]) {
+    var methodParams = [ParameterType]()
+    var optionalParams = [ParameterType]()
+    for param in params ?? [] {
+        // TODO: Virtual Params here!!!
+        if case let .virtual(virt) = param {
+            print("\(virt.name) is Virtual Parameter for \(virt.originalParameter.name)")
```

Comment:
change print() to logger here

Reply:
This line will go away entirely once the PR is ready :)